addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -1657,8 +1657,8 @@ const Wysiwyg = Widget.extend({
                         // Deselect tables so the applied color can be seen
                         // without using `!important` (otherwise the selection
                         // hides it).
-                        if (this.odooEditor.deselectTable() && hasValidSelection(this.odooEditor.editable)) {
-                            this.odooEditor.document.getSelection().collapseToStart();
+                        if (hasValidSelection(this.odooEditor.editable)) {
+                            this.odooEditor.deselectTable();
```

Comment:
If you do that then you can't see the color you just applied since the light blue of the selection hides it. I remember we had a discussion about it with @lebl-odoo but I don't remember what was the decision at the end.

Reply:
When hovering over a color, we apply `!important` to make the hovered color visible without affecting the selection. The `_onColorButtonEnter` method triggers a `historyResetComputedSelection`, which restores the table selection by calling `_selectTableCells`. Additionally, `deselectTable` is called from `_selectTableCells` to ensure the hover color is properly displayed on the cells.

However, when clicking the color button, `!important` is also applied, and the `onColorButtonClick` method is triggered. In `wysiwyg.js` (https://github.com/odoo/odoo/blob/16.0/addons/web_editor/static/src/js/wysiwyg/wysiwyg.js#L1650), this causes the selection to collapse to the start. To prevent the selection from being lost, we can simply call `deselectTable` instead of collapsing the selection.

Reply:
I would like to see I still select a cell (or multiple cells), but I would like to see the background color has changed too. You can have a look of how Google docs does it if you need an example, cc @dmo-odoo & @adch-odoo

Reply:
@lebl-odoo Do you mean something like this
![image](https://github.com/user-attachments/assets/5e7a9e1f-f92f-423c-a154-f8bb977b8fc7)
I'm not sure if we can do that right now in stable. That would be a completely new improvement task, which might be possible using overlays. Google Docs uses a canvas to render the result, which we are not privileged to.
cc. @dmo-odoo

Reply:
@deso-odoo In that case, the minimum is to keep the behavior from web_editor which is : preview the bgcolor, then restore selection once bg color is applied (which seems be the case in the PR). For the text selection, the PR's behavior suits me